build/Band_Representations.jl

```diff
@@ -0,0 +1,310 @@
+            character_table[i, j]=tr(irreps[j].matrices[i]) +0.00im +0.00
+     end
+    end
+character_table
+end
+
+
+"This function provides the induced band representation"
+function induced_band_representation(sitegroup::SiteGroup{D}, h::SymOperation{D}, ρ::Int64, k::KVec) where D
+    orbit_of_sitegroup=orbit(sitegroup, sitegroup.wp);
+    length_of_orbit=length(orbit_of_sitegroup)
+    #Now we sum over all the wyckoff positions in the orbit  
+    gα=cosets(sitegroup)
+    χ_G_k_G=0;
+    wp=sitegroup.wp;
+    wp_rvec=wp.qv.cnst
+    gα_inverse=gα[1]
+
+    transformed_k=transform(k, rotation(h)).cnst
+    for α′ in 1:length_of_orbit 
+        #print("here1\n\n\n")    
+
+        gα′=gα[α′]
+        tα′α′=rotation(h)*orbit_of_sitegroup[α′].qv.cnst+translation(h)-orbit_of_sitegroup[α′].qv.cnst
+        symop_t=SymOperation(one(Crystalline.SquareStaticMatrices.SqSMatrix{D,Float64}), -tα′α′ )
+        
+      #  print("here2\n\n\n")    
+
+        #println(tα′α′)
+        #println(transformed_k)
+        #println(exp(1im*dot(transformed_k,tα′α′)))
+        #find g_alpha_inverse
+
+      # for possible_inverse in 1:length_of_orbit 
+       #     if  gα[α′] ∘ gα[possible_inverse]== spacegroup(17, D)[1] 
+       #        gα_inverse=gα[possible_inverse];
+       #    end
+       #end
+
+        gα_inverse=inv(gα′)    
+
+        #site_symmetry_index=in_site_symmetry(sitegroup, gα_inverse ∘ symop_t ∘ h ∘ gα[α′])
+       site_symmetry_index=in_site_symmetry(sitegroup, compose(gα_inverse, compose(symop_t, compose(h,  gα[α′], false), false ), false)   )
+      
+     #  print("here3\n\n\n")    
+
+       #print(site_symmetry_index, "\n\n\n")
+
+        if site_symmetry_index >0
+            #println(tα′α′)
+            #print(exp(2*pi*1im*dot(transformed_k,tα′α′)))
+            χ_G_k_G=χ_G_k_G+exp(2im*π*dot(transformed_k, tα′α′))*tr(site_symmetry_irreps(sitegroup)[ρ].matrices[site_symmetry_index])
+            #print(transformed_k)
+            #print(dot(transformed_k, tα′α′))
+        end
+    end
+    return χ_G_k_G
+end
+
+
+function subduce_onto_littlegroups(k::String, kvec::KVec, ρ::Int64, sitegroup::SiteGroup{D}) where D
+    sgnum=sitegroup.num
+    num_lgirreps= size(get_lgirreps(sgnum, D)[k])[1]
+    
+    size_lgirreps=size(get_lgirreps(sgnum, D)[k][1].matrices)[1]
+
+    character_table=Matrix{Complex{Float64}}(undef, size_lgirreps, num_lgirreps )
+
+    for j in 1:num_lgirreps
+        for i in 1:size_lgirreps
+            character_table[i, j]=tr(get_lgirreps(sgnum, D)[k][j].matrices[i])
+        end
+    end
+    #return character_table
+    
+    #Now we obtain the vector that we require: 
+    XVector=Matrix{Complex{Float64}}(undef, (size_lgirreps, 1))
+
+    for i in 1:size_lgirreps
+        rep_index=group_representative(group(get_lgirreps(sgnum, D)[k][1])[i], spacegroup(sgnum, D)) 
+        iter=induced_band_representation(sitegroup, spacegroup(sgnum, D)[rep_index], ρ, kvec)
+       XVector[i, 1]= iter
```

Comment:
```suggestion
        XVector[i] = induced_band_representation(sitegroup, spacegroup(sgnum, D)[rep_index], ρ, kvec)
```

Also; much better to define `sg = spacegroup(sgnum, D)` *outside* the loop and then use that; otherwise you are effectively asking the computer to go look in the harddisk every loop iteration, which is very slow.